boards: stm32n6: Update requirements about signing utility (#85399)
zephyrproject-rtos/zephyr: boards/st/nucleo_n657x0_q/doc/index.rst

so its :ref:`installation <stm32cubeprog-flash-host-tools>` is needed.
Version 2.18.0 or later of `STM32CubeProgrammer`_ is required.

.. note::

>    Firmware is run in secure mode of execution, which requires a signature.

Reviewer: Nit:
```suggestion
   Firmware runs in secure mode of execution, which requires a signature.
```
or a more natural
```suggestion
   Firmware is executed in secure mode, which requires a signature.
```

>    West build command is designed to sign generated binary automaticlly using

Author: > `The build system has the ability to sign the generated binary automatically using`

Well, it's a bit more than an ability. It's actually made to sign automatically by default.

>    West build command is designed to sign generated binary automaticlly using

Author: I've reworded. PTAL.

>    After build, the build system  will automatically generate a signed version of the

Reviewer: ```suggestion
   After build, the build system will automatically generate a signed version of the
```

>    binary using STM32CubeProgrammer utility STM32_SigningTool_CLI.

Reviewer 2: ```suggestion
   binary using `STM32CubeProgrammer`_ utility `STM32_SigningTool_CLI`.
```

>    This utility is installed along with STM32CubeProgrammer, but make sure it is

Reviewer 2: ```suggestion
   This utility is installed along with `STM32CubeProgrammer`_, but make sure it is
```

>    available in your PATH variable.

Reviewer 2: ```suggestion
   available in your `PATH` variable.
```

Author: This should probably be ``PATH``
